omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController1_10Test.java

```diff
@@ -110,10 +116,44 @@ public void saveAttachment_shouldUpdateObsComment() throws Exception {
 		assertEquals(editedComment, comment);
 	}
 	
+
+	@Test
+	public void shouldGetAttachment() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		
+		Assert.assertEquals(getUuid(), PropertyUtils.getProperty(result, "uuid"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "comment"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "bytesMimeType"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "bytesContentFamily"));
+	}
+	
+
+	@Test
+	public void shouldGetBytesMimeTypeOfAttachment() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		
+		ComplexData complexData = obs.getComplexData();
+		
+		assertEquals(result.get("bytesMimeType"),complexDataHelper.getContentType(complexData));
```

Comment:
Compare the resulting mime type with what you're expecting. For example if the attachment represents an image then the mime type could be something like `image/png` then your assert should look something like

```java
assertEquals(result.get("bytesMimeType"), "image/png");
```

Do the same thing for the `bytesContentFamily`. 

**Note** the test is not the solution, the test is just to show that the solution works or that it doesn't. In this case the solution doesn't work as I showed you in an earlier screenshot. Which is why I asked you to write this test so it helps you debug the code.

Reply:
@ivange94 i have added the comparison test and also an image attachment of `"image/png"` and it worked as expected

Reply:
@gitcliff can you please test this manually too like I did and confirm that it returns the correct byte type? If it works manually for you, please do share the screenshots and what you did because I might be doing something wrongly as it didn't work for me.

Reply:
Can you try with an image? I tried with an image.